lib/i18n/backend/base.rb

```diff
@@ -236,14 +236,29 @@ def load_rb(filename)
           eval(IO.read(filename), binding, filename)
         end
 
+      YAML_LOAD_OPTIONS = {
+        symbolize_names: true,
+        freeze: true,
+      }.freeze
+
+      # Feature tests YAML load options and only selects those that
+      # are supported in the current version of Pysch.
+      def yaml_load_options
+        @yaml_load_options ||= YAML_LOAD_OPTIONS.select do |k, v|
+          YAML.load_file('test/test_data/locales/en.yml', k => v)
```

Comment:
```suggestion
          YAML.load('', **{k => v})
```

Two things here:
  - That `load_file` wouldn't work once loaded in an app because it is relative to the current working directory (`$PWD)`). But you don't need to load a file anyway, `load` take the same options.
  - It's best to use `**{}` to make sure it's passed as keyword arguments, not a positional hash.

Reply:
Initially, I was trying to use `#load`, but it turns out load's signature wasn't always a good proxy for the signature of `load_file`. 

Until this commit (by yourself), https://github.com/ruby/psych/pull/463, not all kwargs on `load` were forwarded to `load_file`.

This is problematic, as for instance on Ruby 2.7.4, `load` accepts `:symbolize_names` but `load_file` doesn't.

Any suggestions? I certainly agree that using `load` would be optimal, and didn't fully think through the consequence of using `load_file` as you pointed out above.

Reply:
Could accessing the methods args directly help here? Something along the lines of 

```ruby
YAML_LOAD_OPTIONS = YAML.method(:load_file).parameters.each_with_object({}) do |(_context, param_name), args|
  args[param_name] = true if %i[symbolize_names freeze].include?(param_name)
end
```

Reply:
> Could accessing the methods args directly help here?

In this instance no, because for a very long time these methods weren't taking keyword args, but one big option hash.